src/Symfony/Component/DependencyInjection/Tests/ContainerBuilderTest.php

```diff
@@ -63,6 +63,18 @@ public function testDefinitions()
         }
     }
 
+    public function testDeprecatedDefinition()
+    {
+        $definition = new Definition('stdClass');
+        $definition->setDeprecated(true);
+
+        $builder = new ContainerBuilder();
+        $builder->setDefinition('deprecated_foo', $definition);
+
+        //$this->setExpectedException('PHPUnit_Framework_Error', 'The service deprecated_foo relies on a deprecated definition. You should avoid using it.');
```

Comment:
It seems that silent deprecation errors are not detected by phpunit, so the exception is not thrown... wny idea how I could test that ? Thanks